tests/unit/niceml/utilities/test_readwritelock.py

```diff
@@ -20,79 +29,81 @@ def fs_path_config():
 
 
 def test_write_lock(fs_path_config):
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
     write_lock = WriteLock(fs_path_config)
+    assert write_lock.is_acquirable()
     write_lock.acquire()
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
+    assert not write_lock.is_acquirable()
     time.sleep(1)
     write_lock.release()
 
     # lock files should not exist anymore after release
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "read.lock"))
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
+    assert write_lock.is_acquirable()
 
 
-def test_read_write_lock(fs_path_config):
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
-    write_lock = WriteLock(fs_path_config, write=True)
-    write_lock.acquire()
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
-    time.sleep(1)
-    write_lock.release()
+def test_equality(fs_path_config):
+    write_lock1 = WriteLock(fs_path_config)
+    write_lock2 = WriteLock(fs_path_config)
+    assert write_lock1 == write_lock2
 
-    # lock files should not exist anymore after release
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "read.lock"))
-        assert not cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
+    read_lock1 = ReadLock(fs_path_config)
+    read_lock2 = ReadLock(fs_path_config)
+    assert read_lock1 == read_lock2
+
+    assert read_lock1 != write_lock1
+
+    write_lock3 = WriteLock(fs_path_config, write_lock_name="test.lock")
+    read_lock3 = ReadLock(fs_path_config, timeout=10)
+    assert write_lock3 != write_lock1
+    assert read_lock3 != read_lock1
 
 
 def test_retry_lock(fs_path_config):
     # create read lock file manually
     with open_location(fs_path_config) as (cur_fs, root_path):
         cur_fs.touch(join_fs_path(cur_fs, root_path, "write.lock"))
 
-    write = WriteLock(fs_path_config, retry_time=0.1, timeout=1)
+    write_lock = WriteLock(fs_path_config, retry_time=0.1, timeout=1)
     start_time = time.monotonic()
     with pytest.raises(TimeoutError):
-        write.acquire()
+        write_lock.acquire()
     end_time = time.monotonic()
     assert end_time - start_time >= 1
 
     # lock file should still exist
-    with open_location(fs_path_config) as (cur_fs, root_path):
-        assert cur_fs.exists(join_fs_path(cur_fs, root_path, "write.lock"))
-    write.release()
+    assert not write_lock.is_acquirable()
+    write_lock.release()
 
+    # TODO: is this a behaviour we want?
```

Comment:
Yes, event though the write lock was released it should not delete the file because it doesn't belong to it.